wagtail/admin/tests/pages/test_delete_page.py

```diff
@@ -40,6 +41,103 @@ def test_page_delete(self):
         # deletion should not actually happen on GET
         self.assertTrue(SimplePage.objects.filter(id=self.child_page.id).exists())
 
+    @override_settings(WAGTAILADMIN_UNSAFE_PAGE_DELETION_LIMIT=10)
+    def test_confirm_delete_scenario_1(self):
+        # If the number of pages to be deleted are less than
+        # WAGTAILADMIN_UNSAFE_PAGE_DELETION_LIMIT then don't need
+        # for confirmation
+        child_1 = SimplePage(title="child 1", slug="child-1", content="hello")
+        self.child_page.add_child(instance=child_1)
+        child_2 = SimplePage(title="child 2", slug="child-2", content="hello")
+        self.child_page.add_child(instance=child_2)
+        response = self.client.get(
+            reverse("wagtailadmin_pages:delete", args=(self.child_page.id,))
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertNotContains(response, '<input type="text" name="confirm_site_name"')
+        # deletion should not actually happen on GET
+        self.assertTrue(SimplePage.objects.filter(id=self.child_page.id).exists())
+
+        # And admin should be able to delete page without any confirmation
+        response = self.client.post(
+            reverse("wagtailadmin_pages:delete", args=(self.child_page.id,))
+        )
+        # Check that page is deleted
+        self.assertFalse(SimplePage.objects.filter(id=self.child_page.id).exists())
+
+    @override_settings(WAGTAILADMIN_UNSAFE_PAGE_DELETION_LIMIT=3)
+    @override_settings(WAGTAIL_SITE_NAME="mysite")
+    def test_confirm_delete_scenario_2(self):
+        # If the number of pages to be deleted are greater than or equal to
+        # WAGTAILADMIN_UNSAFE_PAGE_DELETION_LIMIT then show input box
+        # to input wagtail_site_name.
+        wagtail_site_name = getattr(settings, "WAGTAIL_SITE_NAME")
+        child_1 = SimplePage(title="child 1", slug="child-1", content="hello")
+        self.child_page.add_child(instance=child_1)
+        child_2 = SimplePage(title="child 2", slug="child-2", content="hello")
+        self.child_page.add_child(instance=child_2)
+        response = self.client.get(
+            reverse("wagtailadmin_pages:delete", args=(self.child_page.id,))
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "This action will delete total <b>3</b> pages.")
+        self.assertContains(
+            response, f"Please type <b>{wagtail_site_name}</b> to confirm."
```

Comment:
A very minor thing... if you're testing for a known string in the response, it's better to give the exact string, with no variables:
```suggestion
            response, "Please type <b>mysite</b> to confirm."
```
That way, there's fewer ways for the test to go wrong - if for some reason WAGTAIL_SITE_NAME was coming through as an empty string in both the test code and the view, this assertion wouldn't spot it.